lib/msf/core/exploit/remote/http/http_cookie_jar.rb

```diff
@@ -0,0 +1,91 @@
+# 3rd party gems
+require 'http/cookie_jar/hash_store'
+require 'http/cookie_jar'
+require 'http/cookie'
+
+module Msf
+  class Exploit
+    class Remote
+      module HTTP
+        # Acts as a wrapper for the 3rd party CookieJar (http-cookie)
+        class HttpCookieJar
+          def initialize
+            @cookie_jar = ::HTTP::CookieJar.new({
+              store: HashStoreWithoutAutomaticExpiration
+            })
+          end
+
+          def add(cookie)
+            raise TypeError, "Passed cookie is of class '#{cookie.class}' and not a subclass of '#{Msf::Exploit::Remote::HTTP::HttpCookie}" unless cookie.is_a?(Msf::Exploit::Remote::HTTP::HttpCookie)
+
+            @cookie_jar.add(cookie)
+            self
+          end
+
+          def delete(cookie)
+            return if @cookie_jar.cookies.empty?
+            raise TypeError, "Passed cookie is of class '#{cookie.class}' and not a subclass of '#{Msf::Exploit::Remote::HTTP::HttpCookie}" unless cookie.is_a?(Msf::Exploit::Remote::HTTP::HttpCookie)
+
+            @cookie_jar.delete(cookie)
+            self
+          end
+
+          # Iterates over all cookies that are not expired in no particular
+          # order.
+          def cookies
+            @cookie_jar.cookies
+          end
```

Comment:
We might revisit this API in the future, as asking the cookie jar for its cookies seems a bit weird - as it already is a collection of cookies by definition 🤔

Reply:
Let's leave this to a follow up PR